frontend/src/pages/shipments/ShipmentForm.tsx

```diff
@@ -0,0 +1,169 @@
+import { gql, useQuery } from '@apollo/client'
+import React, { FunctionComponent, ReactNode, useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import Button from '../../components/Button'
+import SelectField, { SelectOption } from '../../components/forms/SelectField'
+import TextField from '../../components/forms/TextField'
+import { MONTH_OPTIONS } from '../../data/constants'
+import {
+  Group,
+  GroupType,
+  Shipment,
+  ShipmentStatus,
+  ShipmentUpdateInput,
+  ShippingRoute,
+} from '../../types/api-types'
+
+interface Props {
+  /**
+   * If true, the Submit button will be disabled
+   */
+  isLoading: boolean
+  /**
+   * The text to display in the Submit button of the form
+   */
+  submitButtonLabel: ReactNode
+  /**
+   * The values to display in the fields of the form. Note that this is NOT a
+   * controlled component.
+   */
+  defaultValues?: Shipment
+  /**
+   * The callback triggered when the user submits the form
+   */
+  onSubmit: (input: ShipmentUpdateInput) => void
+}
+
+const STATUS_OPTIONS = [
+  { label: 'Abandoned', value: ShipmentStatus.Abandoned },
+  { label: 'Announced', value: ShipmentStatus.Announced },
+  { label: 'Complete', value: ShipmentStatus.Complete },
+  { label: 'In progress', value: ShipmentStatus.InProgress },
+  { label: 'Open', value: ShipmentStatus.Open },
+  { label: 'In staging', value: ShipmentStatus.Staging },
+]
+
+const SHIPPING_ROUTE_OPTIONS = Object.keys(ShippingRoute).map((routeKey) => ({
+  label: (ShippingRoute as any)[routeKey],
+  value: (ShippingRoute as any)[routeKey],
+}))
```

Comment:
I couldn't come up with a more concise way to transform an enum into an object. This works I guess 🙈

Reply:
I could simplify to:
```ts
const SHIPPING_ROUTE_OPTIONS = Object.keys(ShippingRoute).map((routeKey) => ({
  label: routeKey,
  value: routeKey,
}))
```

Reply:
Not sure if this is the proper way to do this, but we could create a helper method for getting enum keys (full context in [this blog post](https://www.petermorlion.com/iterating-a-typescript-enum/)):

```typescript
function enumKeys<O extends object, K extends keyof O = keyof O>(obj: O): K[] {
  return Object.keys(obj) as K[]
}

const SHIPPING_ROUTE_OPTIONS = enumKeys(ShippingRoute).map((routeKey) => ({
  label: ShippingRoute[routeKey],
  value: ShippingRoute[routeKey],
}))
```

This works surprisingly well and fully understands the typing of `routeKey` in context. For example, if I use an invalid key I get a proper error:

<img width="1789" alt="Screen Shot 2021-03-14 at 11 13 29 AM" src="https://user-images.githubusercontent.com/5609090/111079265-7bbf9700-84b6-11eb-88c9-85536ca3c90c.png">

Reply:
Suggestion (blocking): this seems like a great thing to add!

Reply:
Thanks Andrew!